feedservice/handler/handler.go

```diff
@@ -31,23 +31,30 @@ func (handler FeedHandler) On(topic string, message []byte) (*connection.SendRes
 	}
 	if mt.Aggregate == "vt" {
 		switch mt.Function {
-		case "get": // Get single VT metadata
+		case "get": // Get VT Metadata
 			var msg cmds.Get
 			var response messages.Event
 			if err := json.Unmarshal(message, &msg); err != nil {
 				return nil, err
 			}
-			vt, f, err := handler.GetVT(msg)
-			if err != nil {
-				return nil, err
-			}
-			if f != nil {
-				response = f
-			} else {
+			if msg.ID != "" { // Get Single VT
+				vt, err := handler.GetVT(msg.ID)
+				if err != nil {
+					return nil, err
```

Comment:
if you do it like that then the application will crash when the vt wasn't found; a better way to handle this is to create a failure response and return that instead of an error. Like previously in getVt. The reason for that is there are some cases which are legitimate reasons to escalate to the infrastructure (e.g. when no io available while storing) so that the administrator can react to that; however unable to find is not one of those reasons; teherefore creating a failure response and returning that would be better.